pkg/api/api_test.go

```diff
@@ -206,6 +274,10 @@ func getServerConfig(t *testing.T) server.Config {
 		GRPCListenPort:    grpcPortNum,
 
 		GRPCServerMaxRecvMsgSize: 1024,
+
+		// to avoid panic due to
+		// "duplicate metrics collector registration attempted"
+		Registerer: &MockRegisterer{},
```

Comment:
We can simply create new registry. Some duplicate metrics error can point to a real problem, and it's good to catch those.

```suggestion
		// Use new registerer to avoid using default one and panic due to duplicate metrics.
		Registerer: prometheus.NewPedanticRegistry(),
```

We can then remove `MockRegisterer` type.